opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1655,6 +1657,9 @@ namespace Opm
                     FluidSystem::waterPvt().inverseFormationVolumeFactor(pvt_region_index, temperature, seg_pressure, saltConcentration);
                 visc[waterCompIdx] =
                     FluidSystem::waterPvt().viscosity(pvt_region_index, temperature, seg_pressure, saltConcentration);
+                // TODO: double check here
+                // TODO: should not we use phaseIndex here?
```

Comment:
I think I agree with that. The existing code should also be changed. But since it is read in the same way below (by component index) it works for now. A small refactoring of that would be nice in a follow-up PR (please create an issue).